lib/windows/launcher/components/AppManagementView.jsx

```diff
@@ -60,26 +67,85 @@ const AppManagementView = ({
     onReadMore,
     onAppSelected,
     onCreateShortcut,
-}) => getSortedApps(apps).map(app => (
-    <AppItem
-        key={`${app.name}-${app.source}`}
-        app={app}
-        isDisabled={isProcessing}
-        isInstalling={installingAppName === `${app.source}/${app.name}`}
-        isUpgrading={upgradingAppName === `${app.source}/${app.name}`}
-        isRemoving={removingAppName === `${app.source}/${app.name}`}
-        onRemove={() => onRemove(app.name, app.source)}
-        onInstall={() => onInstall(app.name, app.source)}
-        onUpgrade={() => onUpgrade(
-            app.name, app.latestVersion, app.source,
-        )}
-        onReadMore={() => onReadMore(app.homepage)}
-        onAppSelected={() => onAppSelected(app)}
-        onCreateShortcut={() => onCreateShortcut(app)}
-    />
-));
+    onHideReleaseNotes,
+    onShowReleaseNotes,
+}) => (
+    <>
+        {
+            getSortedApps(apps).map(app => (
+                <AppItem
+                    key={`${app.name}-${app.source}`}
+                    app={app}
+                    isDisabled={isProcessing}
+                    isInstalling={installingAppName === `${app.source}/${app.name}`}
+                    isUpgrading={upgradingAppName === `${app.source}/${app.name}`}
+                    isRemoving={removingAppName === `${app.source}/${app.name}`}
+                    onRemove={() => onRemove(app.name, app.source)}
+                    onInstall={() => onInstall(app.name, app.source)}
+                    onReadMore={() => onReadMore(app.homepage)}
+                    onAppSelected={() => onAppSelected(app)}
+                    onCreateShortcut={() => onCreateShortcut(app)}
+                    onShowReleaseNotes={ () => onShowReleaseNotes(
+                        { source: app.source, name: app.name },
+                    )}
+                />
+            ))
+        }
+        {
+            [releaseNotesDialogAppSelection]
+                .filter(x => x.name)
+                .map(({ source, name }) => apps.find(x => x.source === source && x.name === name))
+                .map(({
+                    source,
+                    name,
+                    displayName,
+                    releaseNote,
+                    currentVersion,
+                    latestVersion,
+                }) => (
+                    <Modal
```

Comment:
I would like to suggest to move this modal to be a separate component with a container to lift the logic out from the jsx.